src/csm_testing/tests/sat_functional/bootprep/test_bootprep.py

```diff
@@ -988,6 +989,31 @@ def copy_to_tmp_dir(cls, bootprep_file, dest_folder):
         tmp_bootprep_file_path = os.path.join(cls.temp_dir.name, dest_folder, os.path.basename(bootprep_file))
         shutil.copy(src_bootprep_file_path, tmp_bootprep_file_path)
 
+    @staticmethod
+    def days_since(date_str):
+        """Returns number of days since a given date string
+
+        This handles copying the bootprep input file into the temporary directory
+        and running the command with that directory as the current working directory.
+        Automatically includes --delete-ims-jobs option unless explicitly disabled.
+
+        Args:
+            date_str: date string in the format "2025-12-13T23:37:58.357683"
+
+        Returns:
+            int: number of days since the given date_str or 0 if unable to parse.
+        """
+
+        date_str = date_str.split('T')[0]
+        try:
+            date = datetime.strptime(date_str, '%Y-%m-%d').date()
+            current_date = datetime.today().date()
```

Comment:
The dates appear to be in ISO 8601 format, but we don't have`datetime.fromisoformat` in Python 3.6. Any reason not to keep the whole timestamp though to more accurately calculate days?  Something like:

```suggestion
        try:
            date = datetime.strptime(date_str, '%Y-%m-%dT%H:%M:%S.%f')
            current_date = datetime.now()
```